source/mqtt_agent.c

```diff
@@ -718,6 +729,38 @@ static MQTTStatus_t createAndAddCommand( CommandType_t commandType,
 
 /*-----------------------------------------------------------*/
 
+static void concludeCommand( MQTTAgentContext_t * pAgentContext,
+                             Command_t * pCommand,
+                             MQTTStatus_t returnCode,
+                             uint8_t * pSubackCodes )
+{
+    bool commandReleased = false;
+    MQTTAgentReturnInfo_t returnInfo;
+
+    ( void ) memset( &returnInfo, 0x00, sizeof( MQTTAgentReturnInfo_t ) );
+    assert( pAgentContext != NULL );
+    assert( pCommand != NULL );
+
+    returnInfo.returnCode = returnCode;
+    returnInfo.pSubackCodes = pSubackCodes;
+
+    if( pCommand->pCommandCompleteCallback != NULL )
+    {
+        pCommand->pCommandCompleteCallback( pCommand->pCmdContext, &returnInfo );
+    }
+
+    commandReleased = pAgentContext->agentInterface.releaseCommand( pCommand );
```

Comment:
We assert in mqtt_agent_command_functions.c:239 if this isn't defined, should this function do the same?